### PR TITLE
Added Test option to package add

### PR DESCRIPTION
**Pull request by Bullrich**
Added the option `-t` and `--testables` to add the package name to the `testables` array.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -124,6 +124,12 @@ Turn on debug logging
 openupm --verbose ...
 ```
 
+Add [testables](https://docs.unity3d.com/Manual/cus-tests.html) when importing
+
+```
+openupm --test ...
+```
+
```
Comment (favoyang) [inline README.md:132]: I think these should append to the command-add section, as it only works with it.

File: lib/cli.js [hunk 2]
```diff
@@ -18,7 +18,8 @@ program
   .option("-c, --chdir <path>", "change the working directory")
   .option("-r, --registry <url>", "specify registry url")
   .option("-v, --verbose", "output extra debugging")
-  .option("--no-upstream", "don't use upstream unity registry");
+  .option("--no-upstream", "don't use upstream unity registry")
+  .option("-t", "--test", "add package as testable");
```
Comment (favoyang) [inline lib/cli.js:22]: For the same reason, `-t` should move to the command-add section. See https://github.com/tj/commander.js/issues/521

File: lib/core.js [hunk 3]
```diff
@@ -26,6 +26,7 @@ const parseEnv = function(options, { checkPath }) {
   env.manifestPath = "";
   env.upstream = true;
   env.upstreamRegistry = "https://packages.unity.com";
+  env.testables = false;
```
Comment (favoyang) [inline lib/core.js:29]: Then `core.js` may no longer aware of the testable parameter.

Comment (Bullrich) [reply lib/core.js:29]: Wouldn't it work anyway? We are setting it by default, and it's only being overwritten with the optional value, so it'll always be set.

File: lib/cli.js (hunk 2 above)
Comment (Bullrich) [reply lib/cli.js:22]: My mistake, I fixed it 😄

**Comment by Bullrich** [comment]
Thanks for looking at my PR. I changed the things that you requested and don't know if there is anything it can be done about the `env.testables` parameter

File: lib/core.js (hunk 3 above)
Comment (favoyang) [reply lib/core.js:29]: It "works", but it shouldn't be here. The env contains shared global options for all commands. If `testables` don't make sense for other commands it should not have the default value here, to void confusing or potential conflicting. 

One example is the `deps` command, which has a `--deep` option. The function visit it directly through `options.deep`, which is not part of the `env`.

https://github.com/openupm/openupm-cli/blob/08c3f9bc479b7e0f06b5c7807c245973d09a5e68/lib/cmd-deps.js#L10

**Comment by favoyang** [comment]
> Thanks for looking at my PR. I changed the things that you requested and don't know if there is anything it can be done about the `env.testables` parameter

Thanks for the quick fix! My comment is in https://github.com/openupm/openupm-cli/pull/9#discussion_r441632260

File: lib/core.js (hunk 3 above)
Comment (Bullrich) [reply lib/core.js:29]: Would it be fine if I modify the `_add` method to accept a second optional parameter for testables?

Comment (favoyang) [reply lib/core.js:29]: https://github.com/openupm/openupm-cli/blob/08c3f9bc479b7e0f06b5c7807c245973d09a5e68/lib/cmd-deps.js#L10

Maybe just parse it from the options parameter? Like what the _deps function does.

**Comment by Bullrich** [comment]
I copied the implementation from the deep function. 

Using object destructuring should bring a similar approach. Could you give it a second look please? I don't want to mess with the parameter's object and break the implementation.

**Pull request merged**